src/crimson/os/seastore/seastore_types.h

```diff
@@ -1501,6 +1509,134 @@ constexpr bool is_real_type(extent_types_t type) {
+ */
+struct __attribute__((packed)) lba_map_val_le_t {
+  extent_len_le_t len = init_extent_len_le(0);
+  pladdr_le_t pladdr;
+  extent_ref_count_le_t refcount{0};
+  ceph_le32 checksum{0};
+
+  lba_map_val_le_t() = default;
+  lba_map_val_le_t(const lba_map_val_le_t &) = default;
+  explicit lba_map_val_le_t(const lba_map_val_t &val)
+    : len(init_extent_len_le(val.len)),
+      pladdr(pladdr_le_t(val.pladdr)),
+      refcount(val.refcount),
+      checksum(val.checksum) {}
+
+  operator lba_map_val_t() const {
+    return lba_map_val_t{ len, pladdr, refcount, checksum };
+  }
+};
+
+struct backref_map_val_t {
+  extent_len_t len = 0;	///< length of extents
+  laddr_t laddr = L_ADDR_MIN; ///< logical address of extents
+  extent_types_t type = extent_types_t::ROOT;
+
+  backref_map_val_t() = default;
+  backref_map_val_t(
+    extent_len_t len,
+    laddr_t laddr,
+    extent_types_t type)
+    : len(len), laddr(laddr), type(type) {}
+
+  bool operator==(const backref_map_val_t& rhs) const noexcept {
+    return len == rhs.len && laddr == rhs.laddr;
+  }
+};
+
+std::ostream& operator<<(std::ostream &out, const backref_map_val_t& val);
+
+struct __attribute__((packed)) backref_map_val_le_t {
+  extent_len_le_t len = init_extent_len_le(0);
+  laddr_le_t laddr = laddr_le_t(L_ADDR_MIN);
+  extent_types_le_t type = 0;
+
+  backref_map_val_le_t() = default;
+  backref_map_val_le_t(const backref_map_val_le_t &) = default;
+  explicit backref_map_val_le_t(const backref_map_val_t &val)
+    : len(init_extent_len_le(val.len)),
+      laddr(val.laddr),
+      type(extent_types_le_t(val.type)) {}
+
+  operator backref_map_val_t() const {
+    return backref_map_val_t{len, laddr, (extent_types_t)type};
+  }
+};
+
+class CachedExtent;
+
+/**
+ * ErasedBtreeIter
+ *
+ * ErasedBtreeIter is a wrapper of FixedKVBtree::iterator that povides
+ * more capabilities to operate the LBA/Backref btree compared to
+ * PhysicalNodeMapping. Supported operations are listed in LBAManager
+ * and BackrefManager(TODO).
+ *
+ * The invalidation rule of ErasedBtreeIter:
+ * The iterator becomes invalid when its parent is not viewable(see
+ * CachedExtent::is_viewable_by_trans()) by current transaction. The
+ * btree_iter_version_t field in ErasedBtreeIter records the version
+ * number of btree when it was created. This version is transaction-local
+ * and all insert and remove operations increment this version. The
+ * LBAManager/BackrefManager and FixedKVBtree are responsible for making
+ * sure the iterator is valid.
+ *
+ * 1. When the parent is valid but unviewable by current transaction,
+ *    we could use FixedKVNode::find_pending_version() to find the latest
+ *    pending extent.
+ * 2. When the parent is invalid, start a new lookup to rebuild the iterator.
+ *
+ * When the ErasedBtreeIter is at the end of FixedKVBtree, the key should
+ * be min_max_t<Key>::max.
+ */
+template <typename Key, typename Val>
+struct ErasedBtreeIter {
+  CachedExtent *parent;
```

Comment:
Why it is not a `CachedExtentRef`? Seems to me a raw pointer is dangerous because it may be destructed any time.